src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -26,6 +26,15 @@ import {take} from 'rxjs/operators';
 import {InteractivityChecker} from '../interactivity-checker/interactivity-checker';
 
 
+/** Options for how to handle focus upon opening the dialog */
+export const enum AutoFocusOption {
+  BOTTOM_SHEET = 'bottom-sheet',
+  DIALOG = 'dialog',
+  FIRST_TABBABLE = 'first-tabbable',
+  FIRST_HEADING = 'first-heading',
+  SIDENAV = 'sidenav'
```

Comment:
We don't need to have separate options for bottom-sheet and sidenav. The "dialog" in this context is the ARIA `role="dialog"` attribute, which all of these components use.

Reply:
Maybe we should call it `HOST` instead or `CONTAINER`?

Reply:
I'm inclined to stick with "dialog" since you can explain which element will be focused unambiguously